torch_xla/core/xla_builder.py

```diff
@@ -869,16 +869,17 @@ def jax_func_to_xla_computation(jax_func, args, kwargs, name=None):
   # then the MegaScale device discovery will hang.
   with jax_env_context():
     import jax
-    import torchax.ops.mappings as mappings
+    tx = maybe_get_torchax()
 
     flattened_inputs, spec = jax.tree.flatten((args, kwargs))
 
     def abstractify(a):  # make a pytree leaf abstract
       if a is None:
         return None
       if isinstance(a, torch.Tensor):
-        assert a.device.type == 'xla', f"Inputs must be XLA tensors. Got {a.device}"
-        return jax.ShapeDtypeStruct(a.shape, mappings.t2j_dtype(a.dtype))
+        assert a.device == torch_xla.device(
+        ), f"Inputs must be XLA tensors. Got {a.device}"
+        return jax.ShapeDtypeStruct(a.shape, tx.ops.mappings.t2j_dtype(a.dtype))
```

Comment:
I think this came from an old version of the code. It wouldn't work if the graph was running on `xla:2` etc.

Reply:
https://github.com/pytorch/xla/blob/cce7cc0782bbf507c96e1d81d705db8ec5099f0a/test/test_assume_pure.py#L55 should catch it

Reply:
done.